src/http_server.cc

```diff
@@ -2173,6 +2173,19 @@ HTTPAPIServer::EVBufferToInput(
     }
   }
 
+  // get real header length, only for adsbrain
+  if (n > 0 && typeid(*this) != typeid(HTTPAPIServer)) {
+    header_length = *(unsigned int *)((char *)v[n-1].iov_base + v[n-1].iov_len - 4);
```

Comment:
What will the request body look like? Just curious how the request body from DLIS is handled here.

Reply:
According to c++ doc, typeid output is not fixed, can vary. So a fixed value there may not work someday later. 
The body is constructed by our client, dlis just passes on the data. The body format is:
<json>+<tensor>+<lengthOfJson>

now.

Reply:
it won't break other servers because of line 2177. The other option of modifying req looks heavy because it involves memory allocation and thread lock.

Reply:
Will Line 2177 be true  for any other extended server (e.g., sagemaker_server)?

Reply:
fixed